llvm/include/llvm/IR/DebugInfoMetadata.h

```diff
@@ -2075,6 +2076,11 @@ class DILocation : public MDNode {
   static unsigned
   getBaseDiscriminatorFromDiscriminator(unsigned D,
                                         bool IsFSDiscriminator = false) {
+    // Return the probe id instead of zero for a pseudo probe discriminator.
+    // This should help differenciate callsites with same line numbers.
```

Comment:
Suggest to make it clear that this is so we can still generate good quality AutoFDO profile when -fpseudo-probe-for-profiling is on, in which case we will use probe-id for calls as AutoFDO's discriminator for calls locations to avoid losing all discriminators for callsites.

Reply:
Sounds good, comment updated.